doc/design.md

```diff
@@ -0,0 +1,69 @@
+# Design
+
+
+## Introduction
+The goal of this library is to provide a general framework to implement complex 
+data analysis algorithms with Exasol. The framework basically runs the necessary 
+SQL queries for a given operation in a loop until the state of the operations 
+is completed.
+
+## Constraints / Limitations
+
+- The most efficient way to run SQL queries is to use Lua script.  Only 
+available in Lua Scrips, `pquery` allows SQL queries to be run in the same 
+transactions.
+
+- The results of `pquery` are simple handles and there is a limit of 250 open 
+result set handles for Lua scripts.
+
+
+## Assumptions
+
+- Since the number open result set handles in Lua scripts is limited, it is 
+assumed that Lua scripts is not nested more than 250 times.
+
+- The result of pquery is assumed to not exceed the limit in size.
+
```

Comment:
* The state of the Event Handler is not too large.
* The (de-)serialization and the upload/download of the state to the BucketFS take less than 3 seconds.